pkg/gator/test/test.go

```diff
@@ -51,6 +61,37 @@ func Test(objs []*unstructured.Unstructured) (*GatorResponses, error) {
 			return nil, fmt.Errorf("converting unstructured %q to template: %w", obj.GetName(), err)
 		}
 
+		/** QUESTION FOR MAX:
+		 *
+		 *  If referential data is disabled and a referential template is passed
+		 *  in, an error like the following one is produced:
+		 *
+		 *  ```
+		 *  adding template "k8suniqueserviceselector": invalid
+		 *  ConstraintTemplate: invalid data references: check refs failed on
+		 *  module {template}: errors (1): disallowed ref
+		 *  data.inventory.namespace[namespace][_].Service[name]
+		 *  ```
+		 *
+		 *  I have the option to check for a specific error type in my conditional logic:
+		 *  `frameworks/constraint/pkg/regorewriter.ErrDataReferences`
+		 *
+		 *  I could use that ability to produce a friendlier error message here
+		 *  or in the gator test CLI.  Particularly in gator test CLI, it could
+		 *  be helpful to give a more informative piece of advice to the user.
+		 *  I'm imagining something like:
+		 *
+		 *  "invalid data reference... set --referential-data=true: [rest of error here]"
+		 *
+		 *  Perhaps there's a more typical way to do that.  LMK what you think.
```

Comment:
No objections to a more friendly error message, but we would need to differentiate between attempts to access `data.inventory` and `data.someRandomField`